pkg/v18/resource/certconfig/desired.go

```diff
@@ -118,7 +118,7 @@ func prepareClusterConfig(baseClusterConfig cluster.Config, clusterGuestConfig v
 	if err != nil {
 		return cluster.Config{}, microerror.Mask(err)
 	}
-	clusterConfig.Domain.CalicoEtcdClient = fmt.Sprintf("calico.%s.%s", clusterConfig.ClusterID, key.DNSZone(clusterGuestConfig))
```

Comment:
To make it patch for all version, I would like to avoid using `key` package.

Reply:
The approach we try to take is the WIP should get the shiniest fix. As it gets copied to future versions. It's appreciated that you are backporting this fix. If the backport fix is less pretty but works I'm fine with that.

So does anything need to change here?

Reply:
Nope, we can keep these changes. whenever older release trying to use calico-etcd-client, then we need to fix this anyway.